app/controllers/two_factor_authentication/option_list_controller.rb

```diff
@@ -0,0 +1,46 @@
+module TwoFactorAuthentication
+  class OptionListController < ApplicationController
```

Comment:
What do you think of calling this `LoginOptionsController`, and for the feature for adding additional methods, we'll call it `SetupOptionsController`? If that sounds like a verb, then maybe `OptionsForLoginController` and `OptionsForSetupController`?

Reply:
I wasn't fond of having login in the name since it's already in the path /login/two_factor/XXX  and it makes it redundant.

Reply:
The path name and controller name don't have to match.

Reply:
I'm also a fan of paths that match their associated components.  It makes it easier to find the files.

Reply:
In my other life I developed a routing strategy that doesn't even need the routes file which I believe contributes to the problems of the rails monolith.